pretty_midi/pretty_midi.py

```diff
@@ -461,7 +463,7 @@ def get_end_time(self):
         """
         # Get end times from all instruments, and times of all meta-events
         meta_events = [self.time_signature_changes, self.key_signature_changes,
-                       self.lyrics, self.text_events]
+                       *self.lyrics.values(), *self.text_events.values()]
```

Comment:
Instrument.get_end_time should be modified to take into account lyrics and text events and lyrics/text events should be removed from this list.